In [14]:
import csv
f = open("guns.csv")
data = list(csv.reader(f))
headers = data[0]
data = data[1:]

years = [row[1] for row in data]
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year] = year_counts[year] + 1
    else:
        year_counts[year] = 1
year_counts


{'2012': 33563, '2013': 33636, '2014': 33599}

In [18]:
import datetime
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]
dates[:4]
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1
date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [24]:
sexes = [row[5] for row in data]
sex_counts ={}
for gender in sexes:
    if gender in sex_counts:
        sex_counts[gender] += 1
    else:
        sex_counts[gender] = 1

races = [row[7] for row in data]
race_counts = {}
for race in races:
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1


In [27]:
sex_counts

{'F': 14449, 'M': 86349}

In [28]:
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

# Analysis On Gun Violence From 2012 To 2014

## The most dangerous sex?
Male gun violence is much greater than female violence accounting for 
86349 gun related deaths. Compared to 14449 Female gun realted deaths.

That's around 86% of total gun deaths from Males alone.

## The most dangerous race?
People of the white race topped the count of gun related deaths at 66237,
followed by blacks at 23296; nearly a 1/3. Hispanics following at 9022, asians at 1326 and lastly native ammericans with only 917.

## Further Examinations
At first glance we could presume white males are a major concern when it comes to gun violence. However, before we can jump to any conlusiones we need to look deeper into our data and find the answers to a few specific questions regarding our sample distribution.
1. What is the total race count in our area of gun related deaths, without relation to gun related deaths? In other words how many white people live in the US? How many black people live in the US? And so on. If there is a significat amount of any one race related to the others, surely that can act as a multiplier for gun related deaths related to race. 


In [29]:
f = open("census.csv")
census = list(csv.reader(f))

In [30]:
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [42]:
mapping = {'Asian/Pacific Islander' : 674625+6984195,
          'Black' : 44618105,
          'Native American/Native Alaskan' : 40250635,
          'Hispanic' : 197318956,
          'White' : 308745538}
race_per_hundredk = {}
for i, k in race_counts.items():
    if i in mapping:
        race_per_hundredk[i] = (race_counts[i]/mapping[i])*100000
race_per_hundredk

{'Asian/Pacific Islander': 17.313372033811998,
 'Black': 52.211988832784364,
 'Hispanic': 4.572292588047142,
 'Native American/Native Alaskan': 2.278224927383133,
 'White': 21.453589395678975}

In [56]:
intents = [row[3] for row in data]
races = [row[7] for row in data]
homicide_race_counts = {}
for i, race in enumerate(races):
    if intents[i] == "Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1

for i in homicide_race_counts.keys():
    if i in mapping:
        homicide_race_counts[i] = (homicide_race_counts[i]/mapping[i])*100000

In [57]:
homicide_race_counts

{'Asian/Pacific Islander': 7.298774484842313,
 'Black': 43.7266441503959,
 'Hispanic': 2.8552755975457322,
 'Native American/Native Alaskan': 0.8099251104982568,
 'White': 2.9626339085748987}

# Homicide Rates Proportionate To Population
After filtering our data using the census bureu population data. We are able to measure gun related deaths in proportion to each race in the US.
Since gun related deaths are nominal compared to population numbers, weve multiplied the results by 100,000. Now we can refer to our numbers as x amount of people out of 100,000 died in gun related deaths.

## Between Races
Weve distributed the data in proportion to population and filtered only gun deaths related to homicides. These are the findings:
 'Asian/Pacific Islander': 7.298774484842313,
 'Black': 43.7266441503959,
 'Hispanic': 2.8552755975457322,
 'Native American/Native Alaskan': 0.8099251104982568,
 'White': 2.9626339085748987

We can see that the black population sees more homicides per 100,000 than all other races combined. With Asians coming in far second and whites and Hispanics coming in close for third and fourth place. Native Americans taking the crown with the least homicides per 100,000.

## Next Steps
It would be intersting to see why black homocide rates are so high. Is their another data set we can use to find correlations with homicide rates and cross reference with what we already know? Education, income, location, religion, community involvement, etc.
What are the rates of black homicide rates vs other races and vice versa?
